From 3118958d94890fdaf5d58fd8f84623a09709d45e Mon Sep 17 00:00:00 2001
Message-Id: <3118958d94890fdaf5d58fd8f84623a09709d45e.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Wed, 8 Oct 2014 18:09:09 +0530
Subject: [PATCH 53/75] MUST_REBASE [VPG]: adf/intel/vlv: Do display power
 gating and ungating in DPMS path

During DPMS off display power well will be gated and ungated during DPMS
ON path. S0ix path should be enabled with this

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I71b2439e11d80f8754edc2e97f86362e62b08dd3
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index 1f7c2ad..19bea17 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -154,10 +154,12 @@ static int dsi_dpms(struct intel_pipe *pipe, u8 state)
 
 	switch (state) {
 	case DRM_MODE_DPMS_ON:
+		intel_adf_display_rpm_get();
 		err = vlv_display_on(pipe);
 		break;
 	case DRM_MODE_DPMS_OFF:
 		err = vlv_display_off(pipe);
+		intel_adf_display_rpm_put();
 		break;
 	case DRM_MODE_DPMS_STANDBY:
 	case DRM_MODE_DPMS_SUSPEND:
-- 
1.7.9.5

